### PR TITLE
Ocrvs 2773

### PR DESCRIPTION
**Pull request by emon-swe-sust**
Associated PR on Farajaland: https://github.com/opencrvs/opencrvs-farajaland/pull/232

Icons:
fieldAgentIcon
icon

Pagination:
for showing purpose, set page size 2, when making this gif.
pagination

download fail error message:
errorMessages

After adding Loader ==> 
loader

### CODE REVIEW

File: packages/client/src/components/interface/DownloadButton.tsx [hunk 1]
```diff
@@ -48,18 +48,16 @@ const StatusIndicator = styled.div<{
   isLoading?: boolean
 }>`
   display: flex;
-  flex-grow: 1;
+  flex-grow: 0;
```
Comment (atiqzaman-dsi) [inline packages/client/src/components/interface/DownloadButton.tsx:51]: Why 0 and in core component?

File: packages/client/src/components/interface/DownloadButton.tsx [hunk 2]
```diff
@@ -48,18 +48,16 @@ const StatusIndicator = styled.div<{
...
   align-items: center;
   max-width: 152px;
   justify-content: ${({ isLoading }) =>
     isLoading ? `space-between` : `flex-end`};
 `
-const DownloadAction = styled(TertiaryButton)<{
-  isFullHeight?: boolean
-}>`
-  ${({ isFullHeight }) => isFullHeight && `height: 100%;`}
+const DownloadAction = styled(TertiaryButton)`
```
Comment (atiqzaman-dsi) [inline packages/client/src/components/interface/DownloadButton.tsx:57]: Why is it TertiaryButton? Can't it be IconButton?

File: packages/client/src/utils/authUtils.ts [hunk 3]
```diff
@@ -102,3 +102,34 @@ export function refreshToken() {
...
+  }
+  return false
+}
+
+export const hasRegistrationClerkScope = (scope: Scope | null): boolean => {
+  if (scope?.includes(RouteScope.VALIDATE)) {
+    return true
+  }
+  return false
+}
```
Comment (atiqzaman-dsi) [inline packages/client/src/utils/authUtils.ts:135]: We can simply these logics

File: packages/client/src/utils/authUtils.ts [hunk 4]
```diff
@@ -102,3 +102,34 @@ export function refreshToken() {
...
 }
+
+const enum RouteScope {
+  DECLARE = 'declare',
+  REGISTER = 'register',
+  CERTIFY = 'certify',
+  PERFORMANCE = 'performance',
+  SYSADMIN = 'sysadmin',
+  VALIDATE = 'validate',
+  NATLSYSADMIN = 'natlsysadmin'
```
Comment (atiqzaman-dsi) [inline packages/client/src/utils/authUtils.ts:113]: adding an underscore could be more readable?

File: packages/client/src/views/OfficeHome/LoadingIndicator.tsx [hunk 5]
```diff
@@ -24,7 +24,10 @@ const ErrorText = styled.div`
 `
 const Loading = styled(Spinner)`
   width: 24px;
-  margin-right: 13px;
+  margin-right: auto;
```
Comment (atiqzaman-dsi) [inline packages/client/src/views/OfficeHome/LoadingIndicator.tsx:27]: you can remove this

File: packages/client/src/views/OfficeHome/tabs/WQContentWrapper.tsx [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+                  initialPage={paginationId}
+                  totalPages={totalPages}
+                  onPageChange={onPageChange}
+                />
+              </MobileWrapper>
+            </PaginationWrapper>
+          )}
+        <LoadingIndicator
+          loading={loading ? true : false}
+          hasError={error ? true : false}
```
Comment (atiqzaman-dsi) [inline packages/client/src/views/OfficeHome/tabs/WQContentWrapper.tsx:106]: We can simply logics like loading === true

File: packages/client/src/views/OfficeHome/OfficeHome.test.tsx [hunk 7]
```diff
@@ -230,11 +209,11 @@ describe('OfficeHome related tests', () => {
         app.find('#navigation_readyForReview').hostNodes().text()
       ).toContain('Ready for review3')
       expect(
-        app.find('#navigation_sentForUpdates').hostNodes().text()
-      ).toContain('Sent for updates4')
+        app.find('#navigation_requiresUpdateRegistrar').hostNodes().text()
+      ).toContain('Requires update4')
```
Comment (atiqzaman-dsi) [inline packages/client/src/views/OfficeHome/OfficeHome.test.tsx:213]: what is update4?

File: packages/client/src/views/OfficeHome/tabs/components.tsx [hunk 8]
```diff
@@ -0,0 +1,175 @@
...
+  isValidatedOnReview
+}: IIconWith) => {
+  return (
+    <Flex id="flex">
+      {status && (
+        <IconComp
+          status={status}
+          isDuplicateIcon={isDuplicate}
+          isValidatedOnReview={isValidatedOnReview}
+        />
```
Comment (atiqzaman-dsi) [inline packages/client/src/views/OfficeHome/tabs/components.tsx:100]: Why did you need this IconComp?

File: packages/client/src/views/OfficeHome/tabs/readyToPrint/ReadyToPrint.tsx [hunk 9]
```diff
@@ -0,0 +1,332 @@
...
+  }
+
+  recordWindowWidth = () => {
+    this.setState({ width: window.innerWidth })
+  }
+
+  getExpandable = () => {
+    return this.state.width > this.props.theme.grid.breakpoints.lg
+      ? true
+      : false
```
Comment (atiqzaman-dsi) [inline packages/client/src/views/OfficeHome/tabs/readyToPrint/ReadyToPrint.tsx:103]: Simplify

File: packages/client/src/views/OfficeHome/tabs/readyToPrint/ReadyToPrint.tsx [hunk 10]
```diff
@@ -0,0 +1,332 @@
...
+        title={intl.formatMessage(navigationMessages.print)}
+        isMobileSize={this.state.width < this.props.theme.grid.breakpoints.lg}
+        isShowPagination={isShowPagination}
+        paginationId={paginationId}
+        totalPages={totalPages}
+        onPageChange={onPageChange}
+        loading={this.props.loading}
+        error={this.props.error}
+        noResultText={intl.formatMessage(constantsMessages.noRecords, {
+          tab: 'are ready to print'
```
Comment (atiqzaman-dsi) [inline packages/client/src/views/OfficeHome/tabs/readyToPrint/ReadyToPrint.tsx:305]: need to use intl

File: packages/client/src/views/OfficeHome/tabs/requiresUpdate/RequiresUpdateRegistrar.tsx [hunk 11]
```diff
@@ -223,32 +280,58 @@ class RejectTabComponent extends React.Component<
...
+        title={intl.formatMessage(navigationMessages.requiresUpdateRegistrar)}
+        isMobileSize={this.state.width < this.props.theme.grid.breakpoints.lg}
+        isShowPagination={isShowPagination}
+        paginationId={paginationId}
+        totalPages={totalPages}
+        onPageChange={onPageChange}
+        loading={this.props.loading}
+        error={this.props.error}
+        noResultText={intl.formatMessage(constantsMessages.noRecords, {
+          tab: 'sent for updates'
```
Comment (atiqzaman-dsi) [inline packages/client/src/views/OfficeHome/tabs/requiresUpdate/RequiresUpdateRegistrar.tsx:321]: Use intl

**Review by atiqzaman-dsi — CHANGES_REQUESTED** [review]

File: packages/client/src/components/interface/DownloadButton.tsx (hunk 2 above)
Comment (euanmillar) [reply packages/client/src/components/interface/DownloadButton.tsx:57]: @emon-swe-sust I think using TertiaryButton is dangerous as style may change there

File: packages/client/src/components/interface/DownloadButton.tsx (hunk 1 above)
Comment (euanmillar) [reply packages/client/src/components/interface/DownloadButton.tsx:51]: Agreed but lets merge and then the coming UI dev can move DownloadButton into comp lib

File: packages/client/src/views/OfficeHome/OfficeHome.test.tsx [hunk 12]
```diff
@@ -32,7 +32,7 @@ import { merge } from 'lodash'
 import * as React from 'react'
 
 import { waitFor, waitForElement } from '@client/tests/wait-for-element'
-import { SELECTOR_ID } from './tabs/inProgress/inProgressTab'
+import { SELECTOR_ID } from './tabs/inProgress/InProgress'
```
Comment (euanmillar) [inline packages/client/src/views/OfficeHome/OfficeHome.test.tsx:35]: Please refactor out the "tabs" directory.  The. name "tabs" is confusing.  Just move all files above and let imports update

**Review by euanmillar — CHANGES_REQUESTED** [review]
Refactor out tabs directory

File: packages/client/src/views/OfficeHome/OfficeHome.test.tsx (hunk 7 above)
Comment (emon-swe-sust) [reply packages/client/src/views/OfficeHome/OfficeHome.test.tsx:213]: as we are trying to get the texts only, and there is no space on update and 4(though there is padding betwwen the containers), for this reason, there is no textual space between them.

File: packages/client/src/views/OfficeHome/tabs/components.tsx (hunk 8 above)
Comment (emon-swe-sust) [reply packages/client/src/views/OfficeHome/tabs/components.tsx:100]: on the following to component IconWithName and IconWithNameEvent, i needed IconComp,
